pandas/tests/apply/test_invalid_arg.py

```diff
@@ -207,14 +204,10 @@ def transform(row):
             row["D"] = 7
         return row
 
-    def transform2(row):
-        if notna(row["C"]) and row["C"].startswith("shin") and row["A"] == "foo":
-            row["D"] = 7
-        return row
-
     msg = "'float' object has no attribute 'startswith'"
     with pytest.raises(AttributeError, match=msg):
-        data.apply(transform, axis=1)
+        with tm.assert_cow_warning(warn_copy_on_write):
+            data.apply(transform, axis=1)
```

Comment:
I think the reason I thought this didn't need to warn is that `row` should not be a view, given the dataframe has mixed dtypes. 
So after CoW is enabled, nothing will change here?

Reply:
Yeah sorry, you are correct

The implementation is dirty... I'll mess around with it later today